src/site/content/en/notifications/use-push-notifications-to-engage-users/index.md

```diff
@@ -0,0 +1,162 @@
+
+Bring up the permission prompt **in response to a user gesture** and **when the value of notifications is obvious**. Avoid bringing up the permission prompt immediately when the user lands on the page.
+
+{% endAside %}
+
+## How do push notifications work?
+
+The real power of notifications comes from the combination of service workers and push technology:
+
+*   [Service workers](https://developers.google.com/web/fundamentals/primers/service-workers) can run in the background and display notifications even when your app isn't visible on screen. 
+
+*   Push technology lets you configure your server to send notifications when it makes sense for your app. A push service creates unique URLs for each subscribed service worker. Sending messages to a service worker's URL raises events on that service worker, prompting it to display a notification.
+
+In the following example flow, a client registers a service worker and subscribes to push notifications. Then, the server sends a notification to the subscription endpoint. 
+
+The client and service worker use vanilla JavaScript with no extra libraries. The server is built with the [`express` npm package](https://www.npmjs.com/package/express) on [Node.js](https://nodejs.org/en/), and uses the [`web-push` npm package](https://www.npmjs.com/package/web-push) to send notifications. To send information to the server, the client makes a call to a POST URL that the server has exposed.
+
+### Part 1: Register a service worker and subscribe to Push
+
+1.  A client app registers a service worker with `ServiceWorkerContainer.register()`. The registered service worker will continue to run in the background when the client is inactive. 
+
+    Client code: 
+
+    ```js
+    navigator.serviceWorker.register('sw.js');
+    ```
+
+1.  The client requests permission from the user to send notifications.
+
+    Client code: 
+
+    ```js
+    Notification.requestPermission();
+    ```
+
+1.  To enable push notifications, the service worker uses `PushManager.subscribe()` to create a push subscription. In the call to `PushManager.subscribe()`, the service worker supplies the app's API key as an identifier.
+
+    Client code:
+
+    ```js/0-1/0/
+    navigator.serviceWorker.register('sw.js').then(sw => {
+      sw.pushManager.subscribe({ /* API key */ });
+    });
+    ```
+
+    Push services like Firebase Cloud Messaging operate on a subscription model. When a service worker subscribes to a push service by calling `PushManager.subscribe()`, the push service creates a URL unique to that service worker. The URL is known as a **subscription endpoint**. 
+
+1.  The client sends the subscription endpoint to the app server. 
+
+    Client code:
+
+    ```js/1-3/1/
+    navigator.serviceWorker.register('sw.js').then(sw => {
+      sw.pushManager.subscribe({ /* API key */ }).then(subscription => {
+        sendToServer(subscription, '/new-subscription', 'POST');
```

Comment:
`sendToServer(...)` might be simpler just shown just as a plain old `fetch()` request. To me it looks a bit like some (non-existent) built in function.